src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -0,0 +1,476 @@
+      array_content.push_back(integer2binary(int_value, content_size));
+    }
+    else if(!result[2].str().empty())
+    {
+      char token = result[2].str()[0];
+      array_content.push_back(integer2binary(token, content_size));
+    }
+    else
+    {
+      throw symbol_analysis_exceptiont(
+        "It is not supposed that non group match");
+    }
+    values = result.suffix().str();
+  }
+
+  exprt size = from_integer(array_content.size(), size_type());
+  array_typet new_array_type(type, size);
+  array_exprt result_array =
+    to_array_expr(zero_initializer(new_array_type, location, ns));
+
+  for(size_t i = 0; i < array_content.size(); ++i)
+  {
+    result_array.operands()[i].set(ID_value, array_content[i]);
+  }
+
+  return result_array;
+}
+
+exprt symbol_analyzert::declare_and_initalize_char_ptr(
+  const symbol_exprt &symbol,
+  const std::string &memory_location,
+  const source_locationt &location)
+{
+  const std::regex octal_encoding_pattern("(\\\\[0-9]{3})");
+
+  auto it = values.find(memory_location);
+  if(it == values.end())
+  {
+    std::string value = gdb_api.get_value(id2string(symbol.get_identifier()));
+
+    exprt init = string_constantt(value);
+
+    if(regex_search(value, octal_encoding_pattern))
+    {
+      bitvector_typet bv_type = to_bitvector_type(symbol.type().subtype());
+      init = create_char_array_from_string(value, bv_type, location, ns);
+    }
+
+    code_declt target_object = declare_instance(init.type());
+    target_object.operands().resize(2);
+    target_object.op1() = init;
+    generated_code.add(target_object);
+    const address_of_exprt deref_pointer(target_object.symbol());
+    const exprt casted_if_needed =
+      typecast_exprt::conditional_cast(deref_pointer, symbol.type());
+    values[memory_location] = casted_if_needed;
+    return casted_if_needed;
+  }
+  else
+  {
+    return it->second;
+  }
+}
+
+code_declt symbol_analyzert::declare_instance(const typet &type)
+{
+  const std::string var_id = "id_" + std::to_string(id_counter);
+  ++id_counter;
+  const code_declt declaration(symbol_exprt(var_id, type));
+  return declaration;
+}
+
+exprt symbol_analyzert::process_any_pointer_target(
+  const symbol_exprt &symbol,
+  const std::string memory_location,
+  const source_locationt &location)
+{
+  auto it = values.find(memory_location);
+  if(it == values.end())
+  {
+    const place_holder_exprt recursion_breaker(memory_location);
+    values[memory_location] = recursion_breaker;
+
+    const typet target_type = ns.follow(symbol.type().subtype());
+    const code_declt target_object = get_pointer_target(
+      id2string(symbol.get_identifier()), target_type, location);
+    generated_code.add(target_object);
+    values[memory_location] = target_object.symbol();
+    return target_object.symbol();
+  }
+  else
+  {
+    return it->second;
+  }
+}
+
+code_declt symbol_analyzert::get_pointer_target(
+  const std::string pointer_name,
```

Comment:
Should this really be a `std::string` and not an `irep_idt`?

Reply:
Since we interface gdb, which doesn't really know about ireps, I don't see much added value in using `irep_idt` in some places. Especially if we run `std::string` operations on them afterwards.